### DISCUSSION TITLE
Buckets leave and reenter the bucket list.

### DISCUSSION THREAD

**Discussion by okibcn** [rasa/scoop-directory#93] [General]
@rasa, I see that the directory is not really getting new buckets, but it has a list of about 15-20 buckets bouncing from in and out of consideration every one or two days. How I saw it? well it requires a little intro:
_____
 created a while ago a Scoop search framework for myself. I finally decided to share it with everyone a few days ago given the lack of sound (IMHO) CLI scoop search tool.

The framework has two parts: 
1. A [database generator](https://github.com/okibcn/ScoopMaster) that creates a compact database with all the required info for the search tool.
2. A [search tool](https://github.com/okibcn/ss#ss-scoop-super-search) that gets data from that database.
<img width="1162" alt="image" src="https://user-images.githubusercontent.com/22417711/218010286-09080055-dd27-4f10-bc7d-5d05f504e34c.png">

The database generator is, at this time, unable to explore GitHub to find new buckets. So, it uses Scoop-Directory to retrieve a list of known buckets.
______
After that (long?) intro, you can see in that project the history of two files: [ActiveBuckets](https://github.com/okibcn/ScoopMaster/commits/master/ActiveBuckets) and [DeprecatedBuckets](https://github.com/okibcn/ScoopMaster/commits/master/DeprecatedBuckets). I implemented this tracking a couple of days ago to avoid missing any past buckets ( I am planning on doing the same for apps in the known scoop repos, now tracked as artifacts.).

What is the logic behind the bucket discovery-removal-recover policy in Scoop-Directory?

Thanks

**Comment by rasa** [comment]
@okibcn There is no policy, per se, but we should not drop an included bucket just because GitHub didn't return it in the most recent search result. 

I will consider how much effort it would be to work around this issue. Thanks for bringing it to our attention.

**Reply by okibcn** [reply]
It shouldn't be a big deal. In my case, It was just a line, but it all depends on the code. I haven't gone through your code. Is it the Crawl code in Python forked in other of your GH repos?
